src/cauldrons/WhitelistedCauldronV4.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
This is the same as https://github.com/Abracadabra-money/magic-internet-money/blob/main/contracts/WhitelistedCauldronV3.sol ? But using hooks ?

Reply:
yes exactly

Reply:
makes it cleaner and much easier to upgrade